dispatcher/hub/commonHub.go

```diff
@@ -74,11 +74,13 @@ func checkArgs(args ArgsCommonHub) error {
 }
 
 // Run is launched as a goroutine and listens for events on the exposed channels
-func (ch *commonHub) Run() {
+func (ch *commonHub) Run() error {
 	var ctx context.Context
```

Comment:
we need protection here, also

Reply:
yes, i had that in mind for the next PR when i'll refactor also this file, added a TODO

Reply:
great